csh_auth.go

```diff
@@ -0,0 +1,156 @@
+package csh_auth
```

Comment:
I don't remember which editor you prefer, but for vim `vim-go` and the `go-metalinter` are great tools for catching all the misc style issues, and tiny errors.